packages/commons/src/interop-token/interopTokenService.ts

```diff
@@ -278,4 +279,7 @@ export class InteropTokenGenerator {
 
     return `${serializedToken}.${jwtSignature}`;
   }
+
+  private convertAudience = (input: string | string[]): string | string[] =>
+    Array.isArray(input) && input.length === 1 ? input[0] : input;
```

Comment:
Double check. If the retrieved `descriptorAudience` from the `token-generation-states` table has `length === 1`, it should return a string.